index.html

```diff
@@ -1256,31 +1346,80 @@
   // handle gUM error here
 });
 
-// ... later, in the rendering loop ...
+var gl = canvas.getContext("webgl2");
+// Activate the extension to use single component R32F texture format.
+gl.getExtension('EXT_color_buffer_float');
+
+// Later, in the rendering loop ...
+gl.bindTexture(gl.TEXTURE_2D, depthTexture);
 gl.texImage2D(
    gl.TEXTURE_2D,
    0,
-   gl.RGB,
-   gl.RGB,
-   gl.UNSIGNED_BYTE,
-   depthVideo
-);
+   gl.R32F,
+   gl.RED,
+   gl.FLOAT,
+   depthVideo);
 
-&lt;script id="fragment-shader" type="x-shader/x-fragment"&gt;
-  varying vec2 v_texCoord;
-  // u_tex points to the texture unit containing the depth texture.
-  uniform sampler2D u_tex;
-  uniform float far;
-  uniform float near;
-  void main() {
-    vec4 floatColor = texture2D(u_tex, v_texCoord);
-    float dn = floatColor.r;
-    float depth = 0.;
-    depth = far * near / ( far - dn * ( far - near));
-    // ...
-  }
-&lt;/script&gt;
 </pre>
+      <h3>
+        WebGL2: <dfn>readPixels from float</dfn> texture
+      </h3>
+      <p>
+        This example extends <a>upload to float texture</a> example.
+      </p>
+      <pre class="example">
+// This code creates the texture to which we will upload the depth video frame.
+// Then, it sets up a named framebuffer, attach the texture as color attachment
+// and, after uploading the depth video to the texture, reads the texture
+// content to Float32Array.
+
+// Initialize texture and framebuffer for reading back the texture...
```

Comment:
Please feel free to handle it in your PR #154. My goal with this patch is to replace the original example code - since it is not supported and need to be removed ASAP. Cosmetic changes will need to be handled in follow up.